app/assets/javascripts/add-playlist-option.js.coffee

```diff
@@ -0,0 +1,102 @@
+addnew = 'Add new playlist'
+select_element = $('#post_playlist_id')
+add_success = false
+
+select_element.prepend(new Option(addnew))
+
+getSearchTerm = () ->
+  if $('.select2-search__field')[0]
+    return $('.select2-search__field')[0].value
+  return null
+
+matchWithNew = (params, data) ->
+  term = params.term || ''
+  term = term.toLowerCase()
+  text = data.text.toLowerCase()
+  if (text.indexOf(addnew.toLowerCase()) != -1 || text.indexOf(term) != -1)
```

Comment:
How about using `includes()` instead of `indexOf()` when you don't remember the index of the match?